clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -229,7 +229,7 @@ public long connectionDelay(Node node, long now) {
      */
     @Override
     public boolean connectionFailed(Node node) {
-        return connectionStates.connectionState(node.idString()).equals(ConnectionState.DISCONNECTED);
+        return connectionStates.isDisconnected(node.idString());
```

Comment:
@hachikuji @ijuma I am concerned that maintaining a separate set of nodes only make the code more complex and hard to maintain. Ideally we want to maintain this information in only one state (i.e. unsent) instead of duplicating it in two states.

Maybe we can just define `ConnectionState.DISCONNECTED` to indicate "disconnected after connection" as suggested by Becket. Alternatively, I can also revert to the first version by adding `ClusterConnectionStates.connectionAttempted(...)`. I think either solution is better than maintaining a separate set of nodes that duplicate the information in `unsent`. What do you think?